backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1152,3 +1154,161 @@ func (r *ResourceManager) getNamespaceFromExperiment(references []*api.ResourceR
+					if task.TaskSpec.Volumes == nil {
+						task.TaskSpec.Volumes = []corev1.Volume{}
+					}
+					loggingVolumes := []corev1.Volume{
+						r.getHostPathVolumeSource("varlog", "/var/log"),
+						r.getHostPathVolumeSource("varlibdockercontainers", "/var/lib/docker/containers"),
+						r.getHostPathVolumeSource("varlibkubeletpods", "/var/lib/kubelet/pods"),
+						r.getHostPathVolumeSource("varlogpods", "/var/log/pods"),
+					}
+					task.TaskSpec.Volumes = append(task.TaskSpec.Volumes, loggingVolumes...)
+
+					// Logging volumeMounts
+					if task.TaskSpec.StepTemplate == nil {
+						task.TaskSpec.StepTemplate = &corev1.Container{}
+					}
+					if task.TaskSpec.StepTemplate.VolumeMounts == nil {
+						task.TaskSpec.StepTemplate.VolumeMounts = []corev1.VolumeMount{}
+					}
+					loggingVolumeMounts := []corev1.VolumeMount{
+						{
+							Name:      "varlog",
+							MountPath: "/var/log",
+						},
+						{
+							Name:      "varlibdockercontainers",
+							MountPath: "/var/lib/docker/containers",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "varlibkubeletpods",
+							MountPath: "/var/lib/kubelet/pods",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "varlogpods",
+							MountPath: "/var/log/pods",
+							ReadOnly:  true,
+						},
+					}
+					task.TaskSpec.StepTemplate.VolumeMounts = append(task.TaskSpec.StepTemplate.VolumeMounts, loggingVolumeMounts...)
+				}
+
+				// Upload Artifacts if the artifact is enabled and the annoations are present
+				if hasArtifacts && len(artifacts) > 0 && trackArtifacts {
+					for _, artifact := range artifacts {
+						if len(artifact) == 2 {
+							artifactScript += fmt.Sprintf("push_artifact %s %s\n", artifact[0], artifact[1])
+						}
+					}
+				}
+
+				// Strip EOF if enabled, do it after artifact upload since it only applies to parameter outputs
+				if hasArtifacts && len(artifacts) > 0 && stripEOF {
+					for _, artifact := range artifacts {
+						if len(artifact) == 2 {
+							// TODO: Add EOF newline stripping logics. we need a safe and simple command that can run on the
+							//       minio/mc image. Here we need 1. Check is EOF is newline, 2. Remove EOF newlines and
+							//       update the file. 3. Done with one line bash command. 4. File path is a Tekton substitution
+							//       variable such as $(results.project.path) before runtime.
+							//       We may need to run another image that has python or perl to include these logics.
+							//       This feature is better to implement in Tekton controller since KFP can't inject logics
+							//       during pipeline runtime.
+							//       The below solution is in experimental stage and didn't cover all edge cases.
+							artifactScript += fmt.Sprintf("strip_eof %s %s\n", artifact[0], artifact[1])
+						}
+					}
+				}
+
+				// Define post-processing step
+				step := workflowapi.Step{Container: corev1.Container{
+					Name:  "copy-artifacts",
+					Image: "minio/mc",
```

Comment:
@Tomcli @animeshsingh looking at the current approach I see 2 challenges:
- "minio/mc" image should be customizable - We'd like to use image UBI based on OpenShift
- the auto-generated bash script is still quite hard to customize (even though it is configurable).

Couldn't we simple create a task based on the configurable image (with default e.g.: gcr.io/ml-pipeline/tekton-archiver) that will be responsible for archiving artifacts and logs?

That would give us freedom to replace the archivization logic easily.

Reply:
My original intent was to have a very lightweight image (around 10MB) for pushing artifacts and we don't want to maintain new images since we don't have access to the gcr.io/ml-pipeline CI. With images around 10MB, we won't get full bash shell which is hard to represent some complex data structure, so I just have a simple preprocessing step above to map every artifact into individual sh command. 

However, as we found more gaps in Tekton. I do think these injected extra logics are getting more complex as we are trying do everything on a lightweight kernel. I have a todo at line 1243 to convert everything in this function into a custom Python based image, but it will be at least 2-3x larger so we probably will still let the minio/mc image be our default at the moment.

What I can do for this PR is allowing custom "archiver" image with custom script, so users can provide their own post processing image without going over our custom logics.

Reply:
So I update the image to be customizable, and you can use your own script without going over some of our own post-processing logics. See [here](https://github.com/kubeflow/kfp-tekton/blob/1296cb1c7f86237cd276f7fabd1500e69527f907/kfp-admin-guide.md#customize-artifact-image-to-do-your-own-post-processing) on how to use your own image

Reply:
@rafalbigaj do you think the new changes satisfy your requirements? Let me know is there anything else you want it to be customizable.